### DISCUSSION TITLE
Xonsh on Mac: Python Icon Jumping on the Dock

### DISCUSSION THREAD

**Discussion by ladyrick** [Homebrew/discussions#5748] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.6
ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HEAD: cc0b5d6d05d1f3af5b2f53e9fbb1e7bfa0c332ab
Last commit: 3 days ago
Core tap JSON: 20 Nov 03:34 UTC
Core cask tap JSON: 20 Nov 03:34 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0.1-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
The current git origin is:
  https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/opt/homebrew" remote set-url origin https://github.com/Homebrew/brew

Warning: No Cask quarantine support available: unknown reason.
```

### Description of issue

Please refer to https://github.com/xonsh/xonsh/issues/5641

https://private-user-images.githubusercontent.com/61855675/354166779-dcd10e63-da52-486d-be6f-ed313c91ffcc.mov?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNzM3MjcsIm5iZiI6MTczMjA3MzQyNywicGF0aCI6Ii82MTg1NTY3NS8zNTQxNjY3NzktZGNkMTBlNjMtZGE1Mi00ODZkLWJlNmYtZWQzMTNjOTFmZmNjLm1vdj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDAzMzAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllMDNlODI2YTcwZWZlNDAwNTQ0NmU3YzIxYzY3OTAzNDNiMGRkZDhiOGM2NjcyMzI4MGI3MWY2OGNiYjU2NmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Yvvx_udjMxEZpJZa_f1TBhRbd_St_LFNmnjocEPCJPQ

**Comment by p-linnane** [comment]
I responded in the linked issue. All we do is install `xonsh` via pip into a venv, so we aren't changing anything.

**Reply by p-linnane** [reply]
@cho-m Identified this as a bug in the setproctitle resource when run under macOS 15: https://github.com/dvarrazzo/py-setproctitle/issues/143
